tests/next_tests/unit_tests/type_system_tests/test_type_translation.py

```diff
@@ -166,3 +166,19 @@ def test_invalid_symbol_types():
         type_translation.from_type_hint(typing.Callable[[int], str])
     with pytest.raises(ValueError, match="Invalid callable annotations"):
         type_translation.from_type_hint(typing.Callable[[int], float])
+
+
+@pytest.mark.parametrize(
+    "value,expected",
+    [
+        (common.Dims[IDim, JDim], [IDim, JDim]),
+        (common.Dims[IDim, np.float64], ValueError),
+        (common.Dims["IDim"], ValueError),
+    ],
+)
+def test_generic_variadic_dims(value, expected):
```

Comment:
Just a suggestion, feel free to ignore:
```suggestion
def test_generic_variadic_dims(value, expected_dims):
```